app/controllers/orders_controller.rb

```diff
@@ -16,12 +16,37 @@ def create
     order.save ? success(order) : failure
   end
 
+  def update #this action will have to be hit when a merchant fulfills all items and that action will need a query param ?status="packaged"
+    order = Order.find(params[:id])
+    if cancelled?
+      update_and_cancel(order)
+      #the elsif has not been tested
+    elsif fulfilled?(order)
+      order.update(update_params)
+    elsif params[:status] == "shipped"
+      order.update(update_params)
+      redirect_to admin_path
+    end
```

Comment:
On the refactor can't we just do a series of guard cascading guard clauses?

Reply:
I mean this method seems a little convoluted to me at the moment, maybe we should refactor it?